clients/venice-client/src/main/java/com/linkedin/venice/fastclient/ClientConfig.java

```diff
@@ -263,6 +233,12 @@ private ClientConfig(
     this.projectionFieldValidation = projectionFieldValidation;
     this.longTailRetryBudgetEnforcementWindowInMs = longTailRetryBudgetEnforcementWindowInMs;
     this.harClusters = harClusters;
+    if (instanceHealthMonitor == null) {
+      this.instanceHealthMonitor =
+          new InstanceHealthMonitor(InstanceHealthMonitorConfig.builder().setClient(this.r2Client).build());
```

Comment:
The PR description says that instanceHealthmonitor should be a singleton, but i'm not sure I'm seeing where this is enforced?

Reply:
This will be enforced in the MP layer.

Reply:
How is that?  Also would it make sense though to do it in OS?  Or is there something that prevents that from being a good idea?

Reply:
The idea is to have a global config (independent from client config) to control the behavior of `InstanceHealthMonitor`, which can be done easily in MP layer and it is not straightforward to do it in OS unless we hard-code all the configs for `InstanceHealthMonitorConfig` to constructor a singleton of `InstanceHealthMonitor`.